### ISSUE TITLE
Document build instructions for debian bullseye

### ISSUE THREAD

**Issue by jberryman** [graphia-app/graphia#9]
I'm struggling a bit getting a build on `bullseye`, though I think it should be possible.

This is as far as I've gotten:

```
$ sudo apt install libqt5svg5-dev qtdeclarative5-dev qtbase5-dev
$ # cmake per README succeeds
```

At this point there are runtime errors when doing `build/Graphia`. 

Looks like some progress installing some qt libraries:

```
$ sudo apt install qml-module-qtquick-layouts qml-module-qt-labs-platform qml-module-qtquick-controls qml-module-qtquick-dialogs
```

At this point I get:

```
$ build/Graphia 
QML debugging is enabled. Only use this in a safe environment.
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
Warning: Setting a new default format with a different version or profile after the global shared context is created may cause issues with context sharing.
QQmlApplicationEngine failed to load component
qrc:/qml/main.qml:49:5: Type DocumentUI unavailable
qrc:/qml/DocumentUI.qml:1225:13: Type Transforms unavailable
qrc:/qml/Transform/Transforms.qml:42:5: Type CreateTransformDialog unavailable
qrc:/qml/Transform/CreateTransformDialog.qml:311:33: Type TransformParameter unavailable
qrc:/qml/Transform/TransformParameter.qml:221:5: Type TreeComboBox unavailable
qrc:/qml/Controls/TreeComboBox.qml:21:1: module "QtQuick.Controls" version 2.12 is not installed
```

I'd love to be able to build without messing around with a full Qt distribution

**Comment by jberryman** [comment]
Adding `qml-module-qtquick-controls2` to above and it seems to build and run.

...but I can't seem to load a gml file? After opening file graphia offers to open it with "generic" or "websearch", both of which don't work.

**Comment by timangus** [comment]
I can have a look if you supply the gml file...

**Comment by jberryman** [comment]
[graph.gml.tar.gz](https://github.com/graphia-app/graphia/files/5954115/graph.gml.tar.gz)

Thanks. Does it look as though I'm building and launching graphia correctly?

**Comment by timangus** [comment]
The gml file is malformed in at least 2 ways. Firstly, the root element should be called `graph`, not anonymous i.e.
```
graph
[
  ...
]
```

Not,

```
[
  ...
]
```

Secondly, your node and edge ids are larger than 32bit signed integers, which is not allowed for GML. There may be other issues, but I didn't look further. If you've got to the point where you can load a file, it looks at though you've successfully built Graphia.

**Comment by hoijui** [comment]
I just tested (am using Debian myself), and the current required packages are:

```
qt6-base-dev qt6-declarative-dev qt6-svg-dev qt6-webengine-dev libqt6quicklayouts6 libqt6labsqmlmodels6 libqt6quickcontrols2-6 libqt6quickdialogs2-6  libqt6quickdialogs2utils6 libqt6quickshapes6 libqt6webenginequick6
```

The [related PR](https://github.com/graphia-app/graphia/pull/17) used QT5 libs, as it is from back in 2021.

With these packages installed, the cmake configure step succeeds, but I then get compile errors with current master.
Should I open a new issue for them?

**Comment by timangus** [comment]
> With these packages installed, the cmake configure step succeeds, but I then get compile errors with current master.
> Should I open a new issue for them?

Yes, if you believe it's a problem with Graphia itself, as opposed to your configuration.
